backend/src/main/java/com/redhat/cloud/notifications/db/repositories/BehaviorGroupRepository.java

```diff
@@ -156,7 +166,9 @@ boolean update(String orgId, BehaviorGroup behaviorGroup, boolean isDefaultBehav
             q = q.setParameter("orgId", orgId);
         }
 
-        return q.executeUpdate() > 0;
+        if (q.executeUpdate() == 0) {
+            throw new BadRequestException("Behavior group not updated - probably the behavior group was not part of the orgId");
+        }
```

Comment:
Unless I missed something, the only situation where `q.executeUpdate() == 0` would be `true` is if the current request happened concurrently with another request that deletes the BG. Even if that happened, I don't think we should throw any kind of exception. We could simply do this here:
```suggestion
        q.executeUpdate();
```
WDYT?

Reply:
It would also yield `0` if trying to update a behavior group that is not part of your org id - the update to `getBundleId` might cover that though.

Reply:
anyhow - I think it's fine to fail with an exception if nothing is updated- an update was requested, if nothing was updated there is something wrong.

In this case, the only posibilty is if the orgId and the id don't identify a behavior group - but a previous check was just added. I guess we can omit this check.

Reply:
> It would also yield `0` if trying to update a behavior group that is not part of your org id - the update to `getBundleId` might cover that though.

That can't happen now that you fixed my other comment.